spec/controllers/hyrax/generic_works_controller_spec.rb

```diff
@@ -2,8 +2,93 @@
 #  `rails generate hyrax:work GenericWork`
 require 'rails_helper'
 
-RSpec.describe Hyrax::GenericWorksController do
-  it "has tests" do
-    skip "Add your tests here"
+RSpec.describe Hyrax::GenericWorksController, clean_repo: true do
+  routes { Rails.application.routes }
+  let(:main_app) { Rails.application.routes.url_helpers }
+  let(:hyrax) { Hyrax::Engine.routes.url_helpers }
+  let(:user) { create(:user) }
+
+  before { sign_in user }
+
+  describe '#manifest' do
+    let(:work) { create(:work_with_one_file, user: user) }
+    let(:file_set) { work.ordered_members.to_a.first }
```

Comment:
Since the tests mock out the manifest factory, I think you could mock the work show presenter and not need to create a work, file set, or file objects.  This would speed up the tests a lot and avoid needing to add the fixture files to the repo.  And I think the :clean_repo could be dropped as well.

You can mock methods on the controller using the `controller` object:
`allow(controller).to receive(:presenter).and_return(presenter)`